packages/rum-core/src/domain/action/trackClickActions.ts

```diff
@@ -209,21 +216,42 @@ function startClickAction(
 
 type ClickActionBase = Pick<ClickAction, 'type' | 'name' | 'target' | 'position'>
 
-function computeClickActionBase(event: MouseEventOnElement, actionNameAttribute?: string): ClickActionBase {
+function computeClickActionBase(
+  event: MouseEventOnElement,
+  configuration: RumConfiguration
+): ClickActionBase | undefined {
   const rect = event.target.getBoundingClientRect()
+  const privacyLevel = getNodePrivacyLevel(event.target, configuration.defaultPrivacyLevel)
+
+  // When the node is set to hidden, we do not track click action
+  // Make sure everything has been done before return
+  if (privacyLevel === NodePrivacyLevel.HIDDEN && configuration.enablePrivacyForActionName) {
+    return
+  }
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏I think this condition could be done earlier in `processPointerDown`.  The privacy level could be passed to `computeClickActionBase`. This way no need for this check: https://github.com/DataDog/browser-sdk/blob/f0dacd92dfc320ac3b54f0265194e83e62394152/packages/rum-core/src/domain/action/trackClickActions.ts#L91

Reply:
I moved it to `processPointerDown` but I do not see how we could get rid of the check of `clickActionBase` because if would still be undefined when we skip `hidden` override.